### PR TITLE
Completed post server request

### PR DESCRIPTION
**Pull request by khan-osama**
auth2
auth

### CODE REVIEW

File: api-registration/server/index.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    .then(hashedPassword => {
+      const sql = `
+        insert into "users" ("username", "hashedPassword")
+        values ($1, $2)
+        returning *`;
+      const params = [req.body.username, hashedPassword];
+      db.query(sql, params)
+        .then(result => {
+          const user = result.rows;
+          res.sendStatus(201).json(user);
```
Comment (thebearingedge) [inline api-registration/server/index.js]: Here you mean `.status()`. `.sendStatus()` sends a response with no body.

File: api-registration/server/index.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        });
```
Comment (thebearingedge) [inline api-registration/server/index.js]: Be sure to include a `.catch()` here in case the database query fails.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: api-registration/server/index.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    .then(hashedPassword => {
+      const sql = `
+        insert into "users" ("username", "hashedPassword")
+        values ($1, $2)
+        returning *`;
+      const params = [req.body.username, hashedPassword];
+      db.query(sql, params)
+        .then(result => {
+          const user = result.rows;
+          res.status(201).json(user);
```
Comment (thebearingedge) [inline api-registration/server/index.js]: Sorry, I missed this the first time. The client should receive an object and not an array. Also, the client should not receive the hashed password in the response.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: api-registration/server/index.js (hunk 3 above)
Comment (khan-osama) [reply api-registration/server/index.js]: I'm a little confused as the client isn't receiving the hashed password and it's also an object and not an array. It's in the first screenshot as well.

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**